### PR TITLE
Mautic Legacy Actions Refactor

### PR DESCRIPTION
**Pull request by andrewjschuang**
**UPDATE: a new PR was created with the legacy refactor and new triggers - PR https://github.com/PipedreamHQ/pipedream/pull/3073**

Updated legacy actions:

- [X] add-contact-to-campaign
- [X] clone-campaign
- [X] create-contact
- [X] delete-company
- [X] get-campaign
- [X] get-contact
- [X] list-available-contact-fields
- [X] search-campaigns
- [X] search-companies
- [X] search-contacts
- [X] update-contact 

Deleted legacy action:

- [X] make-api-call

Part 1 of #2903.
Part 2: https://github.com/andrewjschuang/pipedream/pull/3

I will update PR 2 above to merge into `pipedream/master` after this PR is approved and merged.

### CODE REVIEW

File: components/mautic/actions/create-contact/create-contact.mjs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  pick,
+  pickBy,
+} from "lodash-es";
+import { ConfigurationError } from "@pipedream/platform";
+
+export default {
+  key: "mautic-create-contact",
+  name: "Create Contact",
+  description: "Creates a new contact. [See docs](https://developer.mautic.org/#create-contact)",
+  version: "0.2.0",
```
Comment (lcaresia) [inline components/mautic/actions/create-contact/create-contact.mjs:12]:
```suggestion
  version: "0.0.1",
```

File: components/mautic/actions/list-available-contact-fields/list-available-contact-fields.mjs [hunk 2]
```diff
@@ -1,27 +1,19 @@
...
-    return await axios($, {
-      method: "get",
-      url: `${this.mautic.$auth.mautic_url}/api/contacts/list/fields`,
-      headers: {
-        Authorization: `Bearer ${this.mautic.$auth.oauth_access_token}`,
-      },
+    const response = await this.mautic.listContactsFields({
+      $,
     });
+    $.export("$summary", "Succesfully retrieved contacts fields");
```
Comment (lcaresia) [inline components/mautic/actions/list-available-contact-fields/list-available-contact-fields.mjs:16]:
```suggestion
    $.export("$summary", "Successfully retrieved contacts fields");
```

You can check and fix this in other places too.

File: components/mautic/actions/search-campaigns/search-campaigns.mjs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import {
+  pick,
+  pickBy,
+} from "lodash-es";
+
+export default {
+  key: "mautic-search-campaigns",
+  name: "Search Campaigns",
+  description: "Gets a list of campaigns by a search term. [See docs](https://developer.mautic.org/#list-campaigns)",
+  version: "0.2.0",
```
Comment (lcaresia) [inline components/mautic/actions/search-campaigns/search-campaigns.mjs:11]:
```suggestion
  version: "0.0.1",
```

File: components/mautic/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "@pipedream/mautic",
+  "version": "0.3.4",
+  "description": "Pipedream Mautic Components",
+  "main": "mautic.app.js",
```
Comment (lcaresia) [inline components/mautic/package.json:5]:
```suggestion
  "main": "mautic.app.mjs",
```

File: components/mautic/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "@pipedream/mautic",
+  "version": "0.3.4",
```
Comment (lcaresia) [inline components/mautic/package.json:3]:
```suggestion
  "version": "0.0.1",
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Thanks for your contribution!

I requestes just some minor changes.

File: components/mautic/actions/create-contact/create-contact.mjs (hunk 1 above)
Comment (andrewjschuang) [reply components/mautic/actions/create-contact/create-contact.mjs:12]: `create-contact` was updated from `create-contacts` (version `0.1.2`).

File: components/mautic/actions/search-campaigns/search-campaigns.mjs (hunk 3 above)
Comment (andrewjschuang) [reply components/mautic/actions/search-campaigns/search-campaigns.mjs:11]: `search-campaigns` was updated from `list-campaigns` (version `0.1.1`).

File: components/mautic/actions/update-contact/update-contact.mjs [hunk 6]
```diff
@@ -1,274 +1,60 @@
...
+    const fields = contact.fields.all;
+    const props = Object.keys(fields)
+      .filter((k) => fields[k])
+      .reduce((props, k) => {
+        props[k] = {
+          type: "string",
+          label: k,
+          optional: true,
+        };
+        return props;
```
[the remark this answers is not in the document]
Comment (andrewjschuang) [reply components/mautic/actions/update-contact/update-contact.mjs:35]: With `Object.keys` in the line above, I think it is very clear that `k` is `key`.

File: components/mautic/actions/search-companies/search-companies.mjs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import {
+  pick,
+  pickBy,
+} from "lodash-es";
+
+export default {
+  key: "mautic-search-companies",
+  name: "Search Companies",
+  description: "Gets a list of companies by a search term. [See docs](https://developer.mautic.org/#list-contact-companies)",
+  version: "0.3.0",
```
[the remark this answers is not in the document]
Comment (andrewjschuang) [reply components/mautic/actions/search-companies/search-companies.mjs:11]: Same as above.

File: components/mautic/package.json (hunk 5 above)
Comment (andrewjschuang) [reply components/mautic/package.json:3]: The versioning for `package.json` is not the same as for components.

File: components/mautic/actions/create-contact/create-contact.mjs (hunk 1 above)
Comment (lcaresia) [reply components/mautic/actions/create-contact/create-contact.mjs:12]: But it's literally a new action, correct?

It's not will be confusing, a new action start with version `0.2.0`?

If I'm wrong just set this as resolved.

Comment (andrewjschuang) [reply components/mautic/actions/create-contact/create-contact.mjs:12]: It's not a new action. I renamed it while refactoring.

File: components/mautic/package.json (hunk 5 above)
Comment (lcaresia) [reply components/mautic/package.json:3]: Have a doc about the `package.json` versioning?

**Pull request closed without merging**